test/e2e/play_kube_test.go

```diff
@@ -1289,6 +1289,40 @@ var _ = Describe("Podman play kube", func() {
 		Expect(logs.OutputToString()).To(ContainSubstring("hello world"))
 	})
 
+	It("podman pod logs test", func() {
+		SkipIfRemote("podman-remote pod logs -c is mandatory for remote machine")
+		p := getPod(withCtr(getCtr(withCmd([]string{"echo", "hello"}), withArg([]string{"world"}))))
+
+		err := generateKubeYaml("pod", p, kubeYaml)
+		Expect(err).To(BeNil())
+
+		kube := podmanTest.Podman([]string{"play", "kube", kubeYaml})
+		kube.WaitWithDefaultTimeout()
+		Expect(kube).Should(Exit(0))
+
+		logs := podmanTest.Podman([]string{"pod", "logs", p.Name})
+		logs.WaitWithDefaultTimeout()
+		Expect(logs).Should(Exit(0))
+		Expect(logs.OutputToString()).To(ContainSubstring("hello world"))
+	})
+
+	It("podman-remote pod logs test", func() {
+		// -c or --container is required in podman-remote due to api limitation.
+		p := getPod(withCtr(getCtr(withCmd([]string{"echo", "hello"}), withArg([]string{"world"}))))
+
+		err := generateKubeYaml("pod", p, kubeYaml)
+		Expect(err).To(BeNil())
+
+		kube := podmanTest.Podman([]string{"play", "kube", kubeYaml})
+		kube.WaitWithDefaultTimeout()
+		Expect(kube).Should(Exit(0))
+
+		logs := podmanTest.Podman([]string{"pod", "logs", "-c", getCtrNameInPod(p), p.Name})
+		logs.WaitWithDefaultTimeout()
+		Expect(logs).Should(Exit(0))
+		Expect(logs.OutputToString()).To(ContainSubstring("hello world"))
+	})
```

Comment:
Apologies for commenting so late.  I had it on my to-review list but didn't manage before.

I want to ask for more tests in a follow-up PR.  There's a couple of things in the tests that I think need improvement:
* We need more tests and exercise _all_ flags of `podman pod logs` (e.g., --follow). The coverage at the moment is low, so the risk of regressions is high.
* What are the _exact_ semantics of --follow? Shall we follow over the entire lifecycle of the pod (currently we do that) or shall we follow all containers except the infra one? The difference is that --follow will not return unless the pod is stopped.  IMHO we should NOT follow the infra container.
* We need to improve testing the output. Currently, we are only checking for substrings but I don't think that's sufficient to catch future regressions. We need to look closer and match _exact_ output. Since we can get the IDs of containers, we should be able to construct an exact string match, not a substring one.
* Last but not least, we have two log drivers (k8s-file and journald) and I think we should tests both as we do in the container logs tests.  I suggest moving the tests away from `play_kube` and into a new file `pod_logs_test.go`.

Again sorry for not reviewing earlier.

Reply:
@vrothberg Yes i think we should address all these point in few more test cases, I'll add different systems tests addressing these points.